cli/cmd/root.go

```diff
@@ -116,9 +136,15 @@ func runCmd(ctx context.Context, ver config.Version) error {
 	}
 	for _, cmd := range adminCmds {
 		cmd.PersistentFlags().StringVar(&cfg.AdminURL, "api-url", cfg.AdminURL, "Base URL for the admin API")
+		if err := cmd.PersistentFlags().MarkHidden("api-url"); err != nil {
+			panic(err)
+		}
 		cmd.PersistentFlags().StringVar(&cfg.AdminTokenOverride, "api-token", "", "Token for authenticating with the admin API")
 		rootCmd.AddCommand(cmd)
 	}
-
+	// silence usage, usage string will only show up if missing arguments/flags
+	rootCmd.SilenceUsage = true
+	// we want to override some error messages
+	rootCmd.SilenceErrors = true
```

Comment:
nit: Move to top of func where other properties on `rootCmd` (like `Version`) is configured